tests/unit_tests/test_tabular_data.py

```diff
@@ -0,0 +1,352 @@
+import numpy as np
+import pytest
+from pathlib import Path
+import pandas as pd
+from numpy import dtype
+import pyarrow as pa
+
+from mokapot.tabular_data import (
+    TabularDataReader,
+    CSVFileReader,
+    ParquetFileReader,
+    DataFrameReader,
+    ColumnMappedReader,
+    TabularDataWriter,
+    CSVFileWriter,
+    auto_finalize,
+)
+
+
+def test_from_path(tmp_path):
+    reader = TabularDataReader.from_path(Path(tmp_path, "test.csv"))
+    assert isinstance(reader, CSVFileReader)
+
+    reader = TabularDataReader.from_path(Path(tmp_path, "test.parquet"))
+    assert isinstance(reader, ParquetFileReader)
+
+    with pytest.warns(UserWarning):
+        reader = TabularDataReader.from_path(Path(tmp_path, "test.blah"))
+    assert isinstance(reader, CSVFileReader)
+
+
+@pytest.mark.filterwarnings("ignore::pandas.errors.ParserWarning")
+def test_csv_file_reader():
+    # Note: this pin file is kinda "non-standard" which is why I put the
+    # filterwarnings decorator before the test function
+
+    path = Path("data", "phospho_rep1.pin")
+    reader = TabularDataReader.from_path(path)
+    names = reader.get_column_names()
+    expected = [
```

Comment:
I think this could be a dictionary ... just to make it more readable (col x should be an f32 ... instead of position y is called x and position y also has type f32)... maybe in the future ...

Reply:
I fully agree with this. Keeping column names and types as parallel lists was an initial design decision, which turned out to be not so optimal. Dicts would be a lot better. 
Furthermore, I think, it would be best to have the names and types in some kind of "schema" object, which keeps the names as a list and the types as a dict, and can also return the types as list if needed, and also do some operations on the types (e.g. like converting between numpy's dtype and the pyarrow types).